woodwork/logical_types.py

```diff
@@ -694,6 +694,11 @@ class PostalCode(LogicalType):
     backup_dtype = "string"
     standard_tags = {"category"}
 
+    def transform(self, series):
+        if pd.api.types.is_numeric_dtype(series):
```

Comment:
Curious if we should leave this in a try-catch scenario. If a user accidentally passes in a numeric type float like `12345.5`,  pandas will throw a `TypeError`. Might be better for us to catch that and throw a useful message on our end (and include test for it if so)

Reply:
agreed, good catch!

Reply:
This would only apply to pandas since dask does not fail until compute and spark allows for float64 to Int64 conversion by dropping out decimal places.